sharing_portal/views.py

```diff
@@ -949,3 +966,113 @@ def create_supplemental_project_if_needed(request, artifact, project):
             artifact_uuid=artifact["uuid"], project=created_project
         )
         daypass_project.save()
+
+
+@check_edit_permission
+@with_trovi_token
+@login_required
+def create_git_version(request, artifact):
+    errors = False
+    if request.method == "POST":
+        remote_url = request.POST.get("gitRemote")
+        git_ref = request.POST.get("gitRef")
+        # Validate git ref
+        for item in ls_remote(remote_url):
+            if item[0] == git_ref:
+                break
+        else:
+            messages.add_message(
+                request,
+                messages.ERROR,
+                "Either invalid git reference specified, or invalid remote URL",
+            )
+            errors = True
+        if not errors:
+            trovi.create_version(
+                request.session.get("trovi_token"),
+                artifact["uuid"],
+                f"urn:trovi:contents:git:{remote_url}@{git_ref}",
```

Comment:
Yes, it may contain a protocol. I think we discussed content IDs may contain colons for this and zenodo. I did have to add some helper urn functions in trovi to prevent things breaking.